utilities/plotting/forcing_animation.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python
+"""
+@author Nels J. Frazier
+@date 01-02-2024
+@version 0.0.0
+
+Utility to generate animations in gif format in both gridded form
+and catchment aggregated form.  Use forcing_animation --help for
+descriptions of options and required inputs.
+
+"""
+import geopandas as gpd
+import matplotlib.pyplot as plt
+#import PIL
+#PIL.Image.MAX_IMAGE_PIXELS = 188745354+10000
+from matplotlib.animation import FuncAnimation, PillowWriter, FFMpegWriter
+import seaborn as sns
+import xarray as xr
+from functools import partial
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+#import pickle
+import argparse
+from pyogrio import read_dataframe
+
+def plot_catchment_frame(ax, forecast, forcing, catchments, i):
+    t = forcing[i]
+    #each dataset is a slice of time
+    ts = t.time.dt.strftime("%Y-%m-%d %H:%M:%S").item()
+    ax.set_title(forecast+"\n"+ts+"\nPrecipitation", fontsize=20)
+    #don't plot new, update old!!!
+    ax.collections[0].set_array(t.values)
+    print("Plotting time ", i)
+    return ax
+
+from matplotlib.collections import QuadMesh
+def plot_grid_frame(ax, forecast, forcing, cmap, i):
+    #Get time at int index i
+    t = forcing['RAINRATE'][i:i+1]
+    ts = t.time.dt.strftime("%Y-%m-%d %H:%M:%S").item() 
+    ax.axes.set_title(forecast+"\n"+ts+"\nPrecipitation", fontsize=20)
+    #update 
+    ax.set_array(t[0].values)
+    print("Plotting time ", i)
+    return ax
+
+cycles = ["Short", "Medium", "Long"]
+sources = ['HRRR-RAP', 'CFS', 'GFS', 'Other']
+
+parser = argparse.ArgumentParser()
+#TODO add cariable arg
+#TODO add cmap arg
+parser.add_argument('-p', '--preview', help="Plot the first frame and view only", required=False, action="store_true")
+parser.add_argument('-s', '--switch', help="switch the orientation of the colorbar from horizontal to vertical", required=False, action="store_true")
+parser.add_argument('-d', '--dpi', required=False, default=100, help="Resolution to save gif", type=int)
+parser.add_argument("source", choices=sources)
+parser.add_argument("cycle", choices=cycles)
+commands = parser.add_subparsers(dest="command")
+
+h_commands = commands.add_parser("hydrofabric")
+h_commands.add_argument('hydrofabric', metavar="hydrofabric geopackage", type=argparse.FileType('r'))
+h_commands.add_argument('data', metavar="netcdf precip", type=argparse.FileType('r'))
+
+g_commands = commands.add_parser("grid")
+g_commands.add_argument('data', metavar="gridded netcdf precip", type=argparse.FileType('r'))
+
+args = parser.parse_args()
+
+forecast = args.cycle+" Range "+args.source
+crs = ''
+epsg = ''
+if(args.command == 'hydrofabric'):
+    epsg = '5070'
+    crs = ccrs.epsg(epsg)
+    crs = ccrs.PlateCarree()
```

Comment:
Which one is `crs` meant to be?

Reply:
The pyproj 5070 epsg code enforces a projection boundary that clips off the southern part of the CONUS domain, so for now using the PlateCarree.  See comment on line 160.  I'll clean this up a bit, just wanted to get it published in some form.